cfgrammar/src/lib/yacc/parser.rs

```diff
@@ -28,7 +28,6 @@ pub enum YaccGrammarErrorKind {
     IllegalName,
     IllegalString,
     IncompleteRule,
-    DuplicateRule,
```

Comment:
After thinking about it I think we should keep this, but only produce it from `add_rule`, which won't be exercised by the parser.  Since add_rule is only called on the first instance of the rule.

Reply:
nevermind, `add_rule` doesn't actually return anything currently (or check for duplicates), so we'd need to make it return `Result<(), YaccGrammarError>` and do double `self.rules.get(rule_name).is_none()` in `add_rule` and outside it to avoid the error.  So it kind of seems like it is fine as-is?

Reply:
I tend to agree that it's best not to produce this from `add_rule`.